packages/docmanager/src/widgetmanager.ts

```diff
@@ -268,15 +263,12 @@ class DocumentWidgetManager implements IDisposable {
         return true;
       }
       if (result) {
-        this._closeGuard = true;
-        widget.close();
-        this._closeGuard = false;
-        // Dispose of document widgets when they are closed.
         widget.dispose();
```

Comment:
I didn't see the need for the close guard. Why `.close()` the widget if the very next thing to happen is disposing it? Just disposing it seems sufficient to me.